### PR TITLE
Support for TECH files in GDS and DXF imports in HFSS 3D Layout

### LINKED ISSUE #2086: HFSS 3D Layout: Support 'tech file' when GDS file is imported not only 'xml file'
### Description of the feature

Some of users use 'tech file' when GDS file is imported.
Now, PyAEDT support only 'xml file' as layer data but I would like you to be able to set 'tech file' as control_file. 

### Steps for implementing the feature

- Hfss3dLayout.import_gds(
     gds_path, 
     aedb_path=None, 
     control_file=None, 
     set_as_active=True, 
     close_active_project=False)

if  extension of control_file is 'xml':
   read xml file...
elif extension of control_file is 'tech':
   read xml file...

or  add 'contorol_file_type' as argument of method.
if control_file_type is 'xml':
  read xml file...
elif contorol_file_type is 'tech:
  read tech file...

### Useful links and references

_No response_

### PR DESCRIPTION
**Pull request by maxcapodi78**
Fix #2086.

### CODE REVIEW

File: pyaedt/hfss3dlayout.py [hunk 1]
```diff
@@ -1279,13 +1283,14 @@ def import_dxf(self, dxf_path, aedb_path=None, control_file=None, set_as_active=
...
         Parameters
         ----------
-        gds_path : str
+        dxf_path : str
             Full path to the DXF file.
         aedb_path : str, optional
             Full path to the AEDB file.
         control_file : str, optional
-            Path to the XML file with the stackup information. The default is ``None``, in
+            Path to the XML or thech file with the stackup information. The default is ``None``, in
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:1291]:
```suggestion
            Path to the XML or tech file with the stackup information. The default is ``None``, in
```

File: pyaedt/generic/general_methods.py [hunk 2]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
...
 
+def tech_to_control_file(tech_path, unit=None, control_path=None):
+    """Convert a ``.tech`` file to a xml file to be used in gds/dxf import.
+
+    Parameters
+    ----------
+    tech_path : str
+        Full path to the .tech file.
+    unit : str, optional
+        Tech units. If specified in tech file this parameter will not be used. Default is `nm`.
```
Comment (MaxJPRey) [inline pyaedt/generic/general_methods.py:1083]:
```suggestion
        Tech units. If specified in tech file this parameter will not be used. Default is ``"nm"``.
```

**Review by MaxJPRey — APPROVED** [review]
LGTM.

File: pyaedt/generic/general_methods.py [hunk 3]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
     return freq, mag, phase
 
 
+def tech_to_control_file(tech_path, unit=None, control_path=None):
+    """Convert a ``.tech`` file to a xml file to be used in gds/dxf import.
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1076]:
```suggestion
    """Convert a TECH file to an XML file to use in a GDS/DXF import.
```

File: pyaedt/generic/general_methods.py [hunk 4]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
...
+
+    Parameters
+    ----------
+    tech_path : str
+        Full path to the .tech file.
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1081]:
```suggestion
        Full path to the TECH file.
```

File: pyaedt/generic/general_methods.py (hunk 2 above)
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1083]:
```suggestion
        Units for values in the TECH file. If units are specified in the TECH file, this
        parameter is not used. The default is ``nm``.
```

File: pyaedt/generic/general_methods.py [hunk 5]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
...
+    unit : str, optional
+        Tech units. If specified in tech file this parameter will not be used. Default is `nm`.
+    control_path : str, optional
+        Output xml file.
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1085]:
```suggestion
        Path for outputting the XML file.
```

File: pyaedt/generic/general_methods.py [hunk 6]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
...
+
+    Returns
+    -------
+    str
+        Out xml file.
```
Comment (PipKat) [inline pyaedt/generic/general_methods.py:1090]:
```suggestion
        Path for the outputted XML file.
```

File: pyaedt/hfss3dlayout.py [hunk 7]
```diff
@@ -1254,8 +1257,9 @@ def import_gds(self, gds_path, aedb_path=None, control_file=None, set_as_active=
         aedb_path : str, optional
             Full path to the AEDB file.
         control_file : str, optional
-            Path to the XML file with the stackup information. The default is ``None``, in
+            Path to the XML or tech file with the stackup information. The default is ``None``, in
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:1260]:
```suggestion
            Path to the XML or TECH file with the stackup information. The default is ``None``, in
```
Question: when is "Full path to...." used in a description versus "Path to...."? We should be consistent, and I assume that there is a difference.

File: pyaedt/hfss3dlayout.py [hunk 8]
```diff
@@ -1254,8 +1257,9 @@ def import_gds(self, gds_path, aedb_path=None, control_file=None, set_as_active=
...
             which case the stackup is not edited.
+            If `tech` file is provided and layer name starts with `"v"` then is mapped as via layer.
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:1262]:
```suggestion
            If a TECH file is provided and the layer name starts with ``"v"``, the layer is mapped as a via layer.
```
It's unclear here whether the pronoun "it" refers to the TECH file or the layer. I assumed layer, but I'd like you to check.

File: pyaedt/hfss3dlayout.py (hunk 1 above)
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:1291]:
```suggestion
            Path to the XML or TECH file with the stackup information. The default is ``None``, in
```

File: pyaedt/hfss3dlayout.py [hunk 9]
```diff
@@ -1279,13 +1283,14 @@ def import_dxf(self, dxf_path, aedb_path=None, control_file=None, set_as_active=
...
-        gds_path : str
+        dxf_path : str
             Full path to the DXF file.
         aedb_path : str, optional
             Full path to the AEDB file.
         control_file : str, optional
-            Path to the XML file with the stackup information. The default is ``None``, in
+            Path to the XML or thech file with the stackup information. The default is ``None``, in
             which case the stackup is not edited.
+            If `tech` file is provided and layer name starts with `"v"` then is mapped as via layer.
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:1293]:
```suggestion
            If a TECH file is provided and the layer name starts with ``"v"``, the layer
            is mapped as a via layer.
```

File: pyaedt/generic/general_methods.py (hunk 2 above)
Comment (PipKat) [reply pyaedt/generic/general_methods.py:1083]: I'd prefer my rewrite below.

Comment (MaxJPRey) [reply pyaedt/generic/general_methods.py:1083]: For sure @PipKat . Yours is way better.

File: pyaedt/hfss3dlayout.py (hunk 9 above)
Comment (maxcapodi78) [reply pyaedt/hfss3dlayout.py]: Fixed

File: pyaedt/hfss3dlayout.py (hunk 8 above)
Comment (maxcapodi78) [reply pyaedt/hfss3dlayout.py]: Fixed. Tks

File: pyaedt/generic/general_methods.py (hunk 6 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py:1090]: Fixed. Tks

File: pyaedt/generic/general_methods.py (hunk 2 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py]: Fixed. Tks

File: pyaedt/generic/general_methods.py (hunk 5 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py]: Fixed. Tks

File: pyaedt/generic/general_methods.py (hunk 4 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py]: Fixed. Tks

File: pyaedt/generic/general_methods.py [hunk 10]
```diff
@@ -1070,6 +1072,66 @@ def parse_excitation_file(
     return freq, mag, phase
 
 
+def tech_to_control_file(tech_path, unit=None, control_path=None):
```
[the remark this answers is not in the document]
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py]: Fixed. Tks

File: pyaedt/generic/general_methods.py (hunk 2 above)
Comment (maxcapodi78) [reply pyaedt/generic/general_methods.py]: Fixed. Tks

File: pyaedt/hfss3dlayout.py (hunk 1 above)
Comment (maxcapodi78) [reply pyaedt/hfss3dlayout.py]: Fixed. Tks

**Pull request merged**